### PR TITLE
feat: adjust brightness and contrast of image

### LINKED ISSUE #289: Adjust brightness of image
### Is your feature request related to a problem?

I'd like to be able to adjust the brightness of an image.

### Desired solution

Add a new method `adjust_brightness` to `Image` that takes one parameter `factor` and returns the adjusted image. The original should be left unchanged.

Regarding `factor`:
* A factor in (1, inf) should brighten the image
* A factor in [0, 1) should darken the image (0 turns the image completely black)
* A factor of 1 should keep the image unchanged

Because of this:
* Raise an error if the factor is < 0.
* Warn if the factor is equal to 1.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by patrikguempel**
Closes #289, #291.

### Summary of Changes
Added features for adjusting both brightness and contrast

Co-authored-by: daniaHu <129186516+daniaHu@users.noreply.github.com>

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/7.0.4) status: ❌ [ERROR](https://github.com/Safe-DS/Stdlib/actions/runs/5290453188)

| Descriptor  |                                Linter                                 |Files|Fixed|                            Errors                            |Elapsed time|
|-------------|-----------------------------------------------------------------------|-----|----:|--------------------------------------------------------------|------------|
|✅ PYTHON    |[black](https://megalinter.io/7.0.4/descriptors/python_black)          |    2|    1|                                                             0|0.85s       |
|❌ PYTHON    |[mypy](https://megalinter.io/7.0.4/descriptors/python_mypy)            |    2|     |[1](https://github.com/Safe-DS/Stdlib/actions/runs/5290453188)|1.84s       |
|✅ PYTHON    |[ruff](https://megalinter.io/7.0.4/descriptors/python_ruff)            |    2|    1|                                                             0|0.02s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/7.0.4/descriptors/repository_git_diff)|yes  |     |no                                                            |0.0s        |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/5290453188)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: src/safeds/data/image/containers/_image.py [hunk 1]
```diff
@@ -286,9 +287,69 @@ def flip_horizontally(self) -> Image:
...
+        Adjust the brightness of an image.
+
+        Parameters
+        ----------
+        factor: float
+            The brightness factor.
+            1.0 will not change the brightness.
+            Below 1.0 will result in a darker image.
+            Above 1.0 will resolut in a brighter image.
+            Has to be bigger than 0.
```
Comment (PhilipGutberlet) [inline src/safeds/data/image/containers/_image.py:305]: has to be bigger or equal than 0

File: src/safeds/data/image/containers/_image.py [hunk 2]
```diff
@@ -286,9 +287,69 @@ def flip_horizontally(self) -> Image:
...
+        """
+        Adjust Contrast of image.
+
+        Parameters
+        ----------
+        factor: float
+            If factor > 1, increase contrast of image.
+            If factor = 1, no changes will be made.
+            If factor < 1, make image greyer.
+            Has to be bigger than 0.
```
Comment (PhilipGutberlet) [inline src/safeds/data/image/containers/_image.py:335]: Has tto be bigger or equal to 0

**Review by PhilipGutberlet — APPROVED** [review]
good work

**Pull request merged**